### ISSUE TITLE
Enhancement: add templates for issue forms

### ISSUE THREAD

**Issue by xexyl** [ioccc-src/temp-test-ioccc#858]
Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after `jprint` is done I can look at this problem for both the other repo and this one.

## The enhancement itself

As per comments [1615964294](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1615964294), my original comment, Landon's reply at [comment 1617030504](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1617030504) and my reply to that at [comment 1617852422](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1617852422) in issue https://github.com/ioccc-src/temp-test-ioccc/issues/3 we need templates for submitting issues to the IOCCC website.

A useful link for the GitHub issue templates is what I brought up before: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms. An important note they note is that this feature is currently in **beta status** so it is subject to change.

I also noticed just now that there might be a web UI for this which allows you to preview it before pushing it to the repo. I'm not sure if this will work with a pull request or not but if it doesn't I can at least can come up with forms / text. What do you think of that Landon ? Well I guess I'll find out if it's possible and if not I can maybe use the temporary repo I created just for this purpose before and maybe it'll give me the proper files. Well we'll see.

**Comment by lcn2** [comment]
Thanks @xexyl for opening this important issue.

**Comment by xexyl** [comment]
> Thanks @xexyl for opening this important issue.

You're very welcome! The documentation on GitHub will be the key to doing this I think. It appears that for this it's not possible (though maybe I missed it) to do this in the web UI so I'll have to probably test this on my test repo for issues but that is not a problem and is probably preferable anyway as it's more intuitive to me to work at the command line and use `git` itself. But it would be nice too to see a preview. Anyway when it's done it'll be easy to do the same in the other repo.

I have good news in that I slept in two days in a row (though I woke myself up at 3 this morning as I was starting to wake up and I hate the sleepy feeling even when I'm trying to sleep - but this is okay and I got enough sleep even though I am still tired).

I made a couple pull requests including fixing `2004/vik2` but not sure what I'll do next: possibly look at other entries with bugs but even this is not certain I am afraid. Today I think will be a day of easiness but then again I have more time today so it might be that I try and get more done here. We'll see! 

I hope you're having a nice sleep my friend!

**Comment by xexyl** [comment]
There IS something important to this issue though that has to be discussed prior to working on it. The following questions have to be answered.

0. What types of issues should there be?
1. For each type what information should we prompt for?
2. ... maybe other things too but the above two are probably the most important ...

Here are my first thoughts on each.


# Types of issues

I believe that AT LEAST the following issue types should be added:

## Errors / problems with the website

This section should be discussed in issue #3 too!

What does this even mean? I think there are potentially a lot of things here and we probably can't anticipate them all so we probably need something for anything else.

It could be display problem but it could be argued that this is a category on its own, maybe?

As much as I dread it do we want the website to be responsive ? 

What html do we add to the markdowns, if any besides `<br>` at the top ? (If the judges or the author(s)? add html then that should obviously remain.)


## Download problems

This might include wrong meta type being declared so that one can't view the file in the browser.  Is there anything else? For dead links see next item!

# Dead links

This is a difficult task I must go through eventually .. see commit f2aa5e03e6e9ff4713d01a68d9958a4d4d3503c0) which adds this task but it probably would be good to have this in a template anyway as it's entirely possible that some will be missed. 

Dead links means to dead URLs and also means any links in the website itself (I am sure I made typos as I added the many links and would be very very surprised if all is okay as most of the time - almost all the time - I did not copy paste the link or verify that the link is correct). This as well includes the name of the link! There are other things here.

Note that sometimes the Internet Wayback Machine can help identify updated URLs!

Does this one include missing files ? Maybe but maybe not.

# Bugs in code

Of course as we figured out over time some of these that might appear to be bugs are more like features. I still feel bad that I got too carried away with some when fixing most entries to say compile and work but then went too far fixing other issues (though of course in some cases it was deemed worth).

Correct me if I'm wrong but in general a _BUG_ should only be considered a bug if the following conditions are met:

0. If the author notes it as a bug it is a feature not a bug (unless the below applies).
1. If it affects usability it should be considered a bug that should be fixed.
2. Obviously compilation errors should be called a bug (I believe that almost all will be fixed - in fact almost all already are fixed due to my work as well as some work by you and Yusuke - but some will only compile in some platforms as part of the entry itself).
3. If the author provides an alt version that fixes the bug then the entry should not be touched.
4. If the author notes the bug as an actual bug that is a problem it MIGHT be worth fixing. There is a hard one to parse ay nd might not be valid (or is only valid in some cases): if the author says it segfaults without enough args then this is a feature NOT a bug!

Sometimes the fix should be an alt version and sometimes not (I have done both numerous times).

## Anything else

What is it? What do you expect? Maybe why (not sure on this part but it might help in justifying it)?

**Comment by xexyl** [comment]
> See [comment 1623881602](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1623881602) about a caution ⚠️ in making too many categories.
> 
> We don't know how many is too many. That's going to be a judgement call AND it will depend on how easy it will be for someone else to read the list and pick a category.

I agree with all of this too.

**Comment by xexyl** [comment]
> > It could be display problem but it could be argued that this is a category on its own, maybe?
> 
> That could be. One might argue that some of the points suggested in [comment 1623872712](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1623872712) fall under a different category.
> 
I agree that they might be a different category but as you pointed out (and I thought of it too) we have there is a risk that there will be too many types if we're not careful.

> Do keep in mind that people **WILL** misfile bugs: that just happens. Moreover giving people too many bug categories will not only increase the chance of a misfile: some people will just give up if they see too many choices they have to select from.

That's true on all parts perhaps esp the last one.

> 
> Often the fact that someone is even willing to take the time report a problem should be honored. We don't want to make it any harder to report an issue and discourage people from trying.

Agreed.

## 

Of course this doesn't bring us any closer to what to have but we can let it sit for a while and discuss more at the time.

**Comment by lcn2** [comment]
> > > It could be display problem but it could be argued that this is a category on its own, maybe?
> 
> > 
> 
> > That could be. One might argue that some of the points suggested in [comment 1623872712](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1623872712) fall under a different category.
> 
> > 
> 
> I agree that they might be a different category but as you pointed out (and I thought of it too) we have there is a risk that there will be too many types if we're not careful.
> 
> 
> 
> > Do keep in mind that people **WILL** misfile bugs: that just happens. Moreover giving people too many bug categories will not only increase the chance of a misfile: some people will just give up if they see too many choices they have to select from.
> 
> 
> 
> That's true on all parts perhaps esp the last one.
> 
> 
> 
> > 
> 
> > Often the fact that someone is even willing to take the time report a problem should be honored. We don't want to make it any harder to report an issue and discourage people from trying.
> 
> 
> 
> Agreed.
> 
> 
> 
> ## 
> 
> 
> 
> Of course this doesn't bring us any closer to what to have but we can let it sit for a while and discuss more at the time.
> 
> 

Perhaps one category (listed at the bottom) should be an "anything else you think needs fixing" category?

**Comment by xexyl** [comment]
> > > > It could be display problem but it could be argued that this is a category on its own, maybe?
> > 
> > 
> > > 
> > 
> > 
> > > That could be. One might argue that some of the points suggested in [comment 1623872712](https://github.com/ioccc-src/temp-test-ioccc/issues/3#issuecomment-1623872712) fall under a different category.
> > 
> > 
> > > 
> > 
> > 
> > I agree that they might be a different category but as you pointed out (and I thought of it too) we have there is a risk that there will be too many types if we're not careful.
> > > Do keep in mind that people **WILL** misfile bugs: that just happens. Moreover giving people too many bug categories will not only increase the chance of a misfile: some people will just give up if they see too many choices they have to select from.
> > 
> > 
> > That's true on all parts perhaps esp the last one.
> > > 
> > 
> > 
> > > Often the fact that someone is even willing to take the time report a problem should be honored. We don't want to make it any harder to report an issue and discourage people from trying.
> > 
> > 
> > Agreed.
> > 
> > Of course this doesn't bring us any closer to what to have but we can let it sit for a while and discuss more at the time.
> 
> Perhaps one category (listed at the bottom) should be an "anything else you think needs fixing" category?

Definitely. That was basically what I had in mind in the last one I suggested (or proposed).

**Comment by lcn2** [comment]
Hope you are continuing to recover, @xexyl.

More [calc progress](https://github.com/lcn2/calc/issues/99#issuecomment-1742451814).

**Comment by xexyl** [comment]
> Hope you are continuing to recover, @xexyl.
> 

Thanks. Too early yet to know ...

> More [calc progress](https://github.com/lcn2/calc/issues/99#issuecomment-1742451814).

Great!

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
Are we at a stage where this issue could be completed?

The FAQ is something that could be referenced, to add further clarification, in these templates. 

Suggestion: When you link to an FAQ entry, try to use a name tag that isn't of the form _faqX_Y_, but instead use a second name tag that is independent of the FAQ number. 

For example see [FAQ 1.3: How has the IOCCC size limit rule changed over the years?](faq.html#size_rule). 

That FAQ has 2 name tags: "faq1_3", and "size_rule".  In the above link we used the "size_rule" tag to link instead of the "faq1_3" tag. 

Yes, the text of the links still has the "FAQ x_y" in the link title.  However if we are consistent in referring externally to FAQ entries this way, and need to rename them for some reason, we will be able to do a simple `sgit`-style update of the string. 

Moreover in the `faq.md` file itself, we can see which FAQ entries are being externally referenced.  

And should there ever be a reason to remember an FAQ entry, the external links to the non "FAQ x_y" will still work. We would just need to remember the FAQ link titles. And if some external reference was missed in the edits, at least the link would still work.

**Comment by xexyl** [comment]
> # QUESTION
> Are we at a stage where this issue could be completed?
> 
> The FAQ is something that could be referenced, to add further clarification, in these templates.
> 
> Suggestion: When you link to an FAQ entry, try to use a name tag that isn't of the form _faqX_Y_, but instead use a second name tag that is independent of the FAQ number.
> 
> For example see [FAQ 1.3: How has the IOCCC size limit rule changed over the years?](faq.html#size_rule).
> 
> That FAQ has 2 name tags: "faq1_3", and "size_rule". In the above link we used the "size_rule" tag to link instead of the "faq1_3" tag.
> 
> Yes, the text of the links still has the "FAQ x_y" in the link title. However if we are consistent in referring externally to FAQ entries this way, and need to rename them for some reason, we will be able to do a simple `sgit`-style update of the string.
> 
> Moreover in the `faq.md` file itself, we can see which FAQ entries are being externally referenced.
> 
> And should there ever be a reason to remember an FAQ entry, the external links to the non "FAQ x_y" will still work. We would just need to remember the FAQ link titles. And if some external reference was missed in the edits, at least the link would still work.

Well the idea of this was to make it easier to report problems. If you believe that that is no longer necessary then you're welcome to close it.

**Comment by xexyl** [comment]
Or are you asking me to look at doing this maybe?

**Comment by lcn2** [comment]
> Or are you asking me to look at doing this maybe?

You initially wrote:

```
Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
```

So it is still assigned to you? :-)

**Comment by xexyl** [comment]
> > Or are you asking me to look at doing this maybe?
> 
> You initially wrote:
> 
> ```
> Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
> ```
> 
> So it is still assigned to you? :-)

Yes but I thought the other issue is more important? I can certainly look at doing this one if you prefer.

**Comment by lcn2** [comment]
> > > Or are you asking me to look at doing this maybe?
> 
> > 
> 
> > You initially wrote:
> 
> > 
> 
> > ```
> 
> > Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
> 
> > ```
> 
> > 
> 
> > So it is still assigned to you? :-)
> 
> 
> 
> Yes but I thought the other issue is more important? I can certainly look at doing this one if you prefer.

Your original statement when you opened this issue still applies:

```
... It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. ...
```

😉

**Comment by xexyl** [comment]
> > > > Or are you asking me to look at doing this maybe?
> > 
> > > 
> > 
> > > You initially wrote:
> > 
> > > 
> > 
> > > ```
> > 
> > > Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
> > 
> > > ```
> > 
> > > 
> > 
> > > So it is still assigned to you? :-)
> > 
> > 
> > 
> > Yes but I thought the other issue is more important? I can certainly look at doing this one if you prefer.
> 
> Your original statement when you opened this issue still applies:
> 
> ```
> ... It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. ...
> ```
> 
> 😉
> 
> We just mentioned the FAQ link possibility in reviewing open issues.

Then I am confused about the initial question you asked in your comment yesterday. 

Please elaborate? Thanks.

**Comment by lcn2** [comment]
> > > > > Or are you asking me to look at doing this maybe?
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > You initially wrote:
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > So it is still assigned to you? :-)
> > > 
> > > 
> > > Yes but I thought the other issue is more important? I can certainly look at doing this one if you prefer.
> > 
> > 
> > Your original statement when you opened this issue still applies:
> > ```
> > ... It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 😉
> > We just mentioned the FAQ link possibility in reviewing open issues.
> 
> Then I am confused about the initial question you asked in your comment yesterday.
> 
> Please elaborate? Thanks.

Just wondering since issue #3 seems about to be completed, if this issue was ready to start.

**Comment by xexyl** [comment]
> > > > > > Or are you asking me to look at doing this maybe?
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > You initially wrote:
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > Please assign this to me. It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. Maybe after jprint is done I can look at this problem for both the other repo and this one.
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > So it is still assigned to you? :-)
> > > > 
> > > > 
> > > > Yes but I thought the other issue is more important? I can certainly look at doing this one if you prefer.
> > > 
> > > 
> > > Your original statement when you opened this issue still applies:
> > > ```
> > > ... It is lower priority for now of course but I will look at this later on once other things are done or closer to being done. ...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 😉
> > > We just mentioned the FAQ link possibility in reviewing open issues.
> > 
> > 
> > Then I am confused about the initial question you asked in your comment yesterday.
> > Please elaborate? Thanks.
> 
> Just wondering since issue #3 seems about to be completed, if this issue was ready to start.

I'll tackle this after that one is done. Which is funny you say that because I noted that fact over there a bit ago.

Thanks.

**Comment by lcn2** [comment]
We raised the priority of this issue because when the news announcement is posted to the main web site, some people will examine this repo and perhaps open up issues that they discoverer, or even submit potential bug fixes and other pull requests.

We need to have the templates ready for when the news announcement is posted.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
# QUESTION

What is the status of this issue?

Do you have an guesstimated time this issue will close?

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
> # QUESTION
> What is the status of this issue?
> 
> Do you have an guesstimated time this issue will close?

I am afraid I don't because I'm working on the other issues first and during recovery (esp) I am quite slow. Another reason I'm unsure about this one is because I'm not sure how to do it yet.

**Comment by xexyl** [comment]
> > > # QUESTION
> 
> > > What is the status of this issue?
> 
> > > Do you have an guesstimated time this issue will close?
> 
> > 
> 
> > I am afraid I don't because I'm working on the other issues first and during recovery (esp) I am quite slow.
> 
> 
> 
> We both understand and appreciate why things are, as you say, _quite slow_, and we wish to good process on your recovery.
> 
> 

Appreciate it very much!

> 
> We only ask [questions in comment-2110450309](https://github.com/ioccc-src/temp-test-ioccc/issues/858#issuecomment-2110450309)  for planning purposes.
> 
> 
I see. Thanks!

> 
> > Another reason I'm unsure about this one is because I'm not sure how to do it yet.
> 
> 
> 
> We recommend that, when you are able, review [the top TODO comment](https://github.com/ioccc-src/temp-test-ioccc/issues/858#issue-1789457201) and related links as a start.  From there we suggest you try to ask specific questions in comments.

Iirc that's for the topics to include. But maybe there is more. I am going for the other issues first since this one I have to research how to do it and I don't have the energy for that. 

But if this is something you want done before the other issues please let me know and I will look at it when I can even if I haven't finished the other issues. Feel free too to remind me of it.

**Comment by xexyl** [comment]
Does the pull request I just opened resolve this issue?

I still think that the wording can be improved but I think the categories are good?

.. unless I made a syntax error of course.  


Please do feel free to update the descriptions or add additional fields that you think are necessary!

I think we do not want a bugs issue after all: after all the code is not maintained and anyone who wants to look at such and try and fix it is welcome to. 

Or perhaps it should be there so it can (if necessary) be added to the bugs file?

I don't think that I will do anything else here today but I hope that this issue can be marked resolved and then there will be one more thing completed!

Tomorrow I can look at the manifest. This issue here was easier than I expected. I was able to look into the 'how' when I was not at the computer and then when back I did some test commits in a repo specifically made to test these things. 

Anyway I do believe that the text can be improved but I think that the issue can also be considered resolved otherwise. 

Though I only tested with one file I think it should be good. Again unless I made a syntax error (which happened in my test as using `o` in vim added tabs and apparently that's not allowed).

**Comment by xexyl** [comment]
> Does the pull request I just opened resolve this issue?
> 
> I still think that the wording can be improved but I think the categories are good?
> 
> .. unless I made a syntax error of course.  
> 
> 
> Please do feel free to update the descriptions or add additional fields that you think are necessary!
> 
> I think we do not want a bugs issue after all: after all the code is not maintained and anyone who wants to look at such and try and fix it is welcome to. 
> 
> Or perhaps it should be there so it can (if necessary) be added to the bugs file?
> 
> I don't think that I will do anything else here today but I hope that this issue can be marked resolved and then there will be one more thing completed!
> 
> Tomorrow I can look at the manifest. This issue here was easier than I expected. I was able to look into the 'how' when I was not at the computer and then when back I did some test commits in a repo specifically made to test these things. 
> 
> Anyway I do believe that the text can be improved but I think that the issue can also be considered resolved otherwise. 
> 
> Though I only tested with one file I think it should be good. Again unless I made a syntax error (which happened in my test as using `o` in vim added tabs and apparently that's not allowed). 
> 
> 

But on the other hand the other repo should probably have a bug report template and also a feature request template. 

The former will take a bit more time as I would have to add details we have already which includes running the bug report script that I wrote. 

I am not sure if I am up to that today: I slept in more than I would like and I have other things I want to do. 

But perhaps it is better to wait for the pull request here to be merged and then we can discuss it, refine as necessary and then I can add it to the other repo. At that point we can discuss further details for that repo over there. 

So I am right now just on the exercise bike but after that I will be doing other things. 

I hope and believe that this issue can be marked resolved unless we want to add another form (one suggestion in previous comment that I am replying to) or we wish to reword things or add to one or more of them before it is marked completed but even so I think it's almost there if not completely. 

Let me know what you think about whether we need any other categories and also what you think of the current ones once you see them rendered here on GitHub. 

I was kind of in a rush at the end but like I said I had done some tests and it looked good: even if some wording can be improved (I don't know but I think the one that rather could be improved is the one that is not a specific issue but rather an open one .. not even sure what to call that one or if we might need more information or not though that latter part also applies to the other ones). 

Anyway depending on when you get to this today and what I am in the middle of I will either reply tomorrow or else today but unless it's a quick reply (or something that will allow us to mark the issue complete) I probably won't get to it until tomorrow.

**Comment by lcn2** [comment]
> But perhaps it is better to wait for the pull request here to be merged and then we can discuss it, refine as necessary and then I can add it to the other repo. At that point we can discuss further details for that repo over there.

We agree, so we have completed the pull request so that one may evaluate the result. 

Thanks for this work, @xexyl.

Let's evaluate the results and see if anything more needs to be done.

**Comment by xexyl** [comment]
> > But perhaps it is better to wait for the pull request here to be merged and then we can discuss it, refine as necessary and then I can add it to the other repo. At that point we can discuss further details for that repo over there.
> 
> We agree, so we have completed the pull request so that one may evaluate the result. 
> 
> Thanks for this work, @xexyl.
> 
> Let's evaluate the results and see if anything more needs to be done. Nevertheless this issue appears to be neatly resolved 👍

I think so too other than as noted numerous times I think that the blank issue one might need better wording. But then perhaps some of the others also need something added that I am unaware of?

I will take a look at the forms soon too.

**Comment by xexyl** [comment]
One thing that I should change is the word 'file' to 'report'. 

I am not sure if I will do that today or if it will be tomorrow but I will. That word was from their example. Much I changed and I added fields too. But some I kept the same or close to the same. It doesn't look good to me though. 

Mostly it does seem okay though I think. What do you think?

**Comment by lcn2** [comment]
> Mostly it does seem okay though I think. What do you think?

Does seem mostly okay. 

We need to look at the issues on a desktop and ponder.

**Comment by lcn2** [comment]
> One thing that I should change is the word 'file' to 'report'.
> 
> I am not sure if I will do that today or if it will be tomorrow but I will. That word was from their example. Much I changed and I added fields too. But some I kept the same or close to the same. It doesn't look good to me though.

This sort of tweaking that you are suggesting might be in order.

**Comment by xexyl** [comment]
> > One thing that I should change is the word 'file' to 'report'.
> > I am not sure if I will do that today or if it will be tomorrow but I will. That word was from their example. Much I changed and I added fields too. But some I kept the same or close to the same. It doesn't look good to me though.
> 
> This sort of tweaking that you are suggesting might be in order.

Okay so file -> report. If you tell me what else you think should change I can do it in a single commit to finish this issue off.

Or if you wish you could of course do it. Let me know.

**Comment by xexyl** [comment]
I believe that commit https://github.com/ioccc-src/temp-test-ioccc/pull/2343/commits/d95d2234d5dbc70e06c862c47f8575e1f4b75d0c will have resolved this issue. It seems pretty nice. Of course it'll be easier for you to see once you've merged and looked at the issue forms themselves. But I hope that this can be marked complete.

I will be doing other things today but if this can be marked complete that would be a top priority issue resolved. Tomorrow, assuming that this issue is done, I will start looking at the manifest I think. For now I am going to do other things, mostly read: a certain book is calling me.

**Comment by lcn2** [comment]
> I believe that commit https://github.com/ioccc-src/temp-test-ioccc/pull/2343/commits/d95d2234d5dbc70e06c862c47f8575e1f4b75d0c will have resolved this issue. It seems pretty nice. Of course it'll be easier for you to see once you've merged and looked at the issue forms themselves. But I hope that this can be marked complete.
> 
> 
> 
> I will be doing other things today but if this can be marked complete that would be a top priority issue resolved. Tomorrow, assuming that this issue is done, I will start looking at the manifest I think. For now I am going to do other things, mostly read: a certain book is calling me.

From the edits things look like they are in good shape for this issue.

Before we consider closing this issue:

We want to look at things on the laptop now that they have merged and ponder over the set that is there now.

**Comment by xexyl** [comment]
> > I believe that commit https://github.com/ioccc-src/temp-test-ioccc/pull/2343/commits/d95d2234d5dbc70e06c862c47f8575e1f4b75d0c will have resolved this issue. It seems pretty nice. Of course it'll be easier for you to see once you've merged and looked at the issue forms themselves. But I hope that this can be marked complete.
> > 
> > 
> > 
> > I will be doing other things today but if this can be marked complete that would be a top priority issue resolved. Tomorrow, assuming that this issue is done, I will start looking at the manifest I think. For now I am going to do other things, mostly read: a certain book is calling me.
> 
> From the edits things look like they are in good shape for this issue.
> 
> Before we consider closing this issue:
> 
> We want to look at things on the laptop now that they have merged and ponder over the set that is there now.

Good idea! Let me know. Feel free to close it if you think it's resolved. 

If it is I will look at the manifest tomorrow. Otherwise I will see about finishing this issue up. Hopefully this is resolved though: would be great to have a high priority issue done. 

Have a great day!

**Comment by lcn2** [comment]
We very much like the 3 new templates.

We suggest 1 more template having to do with correcting information about an author.

Here are some general suggestions for that template (and this is why this isn't a simple website change):

- Ask for the name of the author

- Ask if they are the author (checkbox?)

NOTE: It is OK if they are changing information about someone else/
NOTE: You might want to state that it is OK if they are correcting information about someone else.

- Ask for the JSON filename under the author directory that they are changing

- Ask for the line(s) in the author JSON file they wish to change

- Ask for the what those author JSON file line(s) should be changed into

NOTE: Indicate that if they want to remove some information about an author, they should set the value to _null_ without quotes.

- Ask for how we might be able to verify that their proposed change is correct

- Ask for any additional remarks that might be helpful

**Comment by xexyl** [comment]
> We very much like the 3 new templates.
> 
> 
> 
> We suggest 1 more template having to do with correcting information about an author.
> 
> 
> 
> Here are some general suggestions for that template (and this is why this isn't a simple website change):
> 
> 
> 
> - Ask for the name of the author
> 
> 
> 
> - Ask if they are the author (checkbox?)
> 
> 
> 
> NOTE: It is OK if they are changing information about someone else/
> 
> NOTE: You might want to state that it is OK if they are correcting information about someone else.
> 
> 
> 
> - Ask for the JSON filename under the author directory that they are changing
> 
> 
> 
> - Ask for the line(s) in the author JSON file they wish to change
> 
> 
> 
> - Ask for the what those author JSON file line(s) should be changed into
> 
> 
> 
> NOTE: Indicate that if they want to remove some information about an author, they should set the value to _null_ without quotes.
> 
> 
> 
> - Ask for how we might be able to verify that their proposed change is correct
> 
> 
> 
> - Ask for any additional remarks that might be helpful

Will do! 

I was planning on looking at the manifest today but perhaps I will do the tomorrow instead if I have the time (if I don't I should Tuesday).

**Comment by xexyl** [comment]
> > We very much like the 3 new templates.
> > We suggest 1 more template having to do with correcting information about an author.
> > Here are some general suggestions for that template (and this is why this isn't a simple website change):
> > 
> > * Ask for the name of the author
> > * Ask if they are the author (checkbox?)
> > 
> > NOTE: It is OK if they are changing information about someone else/
> > NOTE: You might want to state that it is OK if they are correcting information about someone else.
> > 
> > * Ask for the JSON filename under the author directory that they are changing
> > * Ask for the line(s) in the author JSON file they wish to change
> > * Ask for the what those author JSON file line(s) should be changed into
> > 
> > NOTE: Indicate that if they want to remove some information about an author, they should set the value to _null_ without quotes.
> > 
> > * Ask for how we might be able to verify that their proposed change is correct
> > * Ask for any additional remarks that might be helpful
> 
> Will do!
> 
> I was planning on looking at the manifest today but perhaps I will do the tomorrow instead if I have the time (if I don't I should Tuesday).

Does commit https://github.com/ioccc-src/temp-test-ioccc/pull/2344/commits/988acebbd5b31d7d4e6bafb4609e72342e729dca resolve this problem ?

I might be able to do a bit with the manifest but I'm not sure yet ... I'm quite tired today and if I'm to go through every line it might be better to have more energy. Though I'm still waking up so it's possible I might be able to do some in a bit. Still not sure how to filter the lines to help out without modifying it but I'll figure it out. I had the idea that maybe making a copy of it and then doing the filtering in the copy, one change at a time, and then updating the actual changes in the tmp/manifest.numbers might be good. I'll have to see.

**Comment by lcn2** [comment]
Well done, @xexyl .. much better than what our approach might have been.  Thank you!

**Comment by xexyl** [comment]
> Well done, @xexyl .. much better than what our approach might have been.  Thank you! 

You are most welcome! I made a temporary test repo to get things right. Good thing I did because there were some syntax errors at some times (though syntax highlighting didn't seem to show any errors nevertheless GitHub had some problems - sometimes with almost entirely useless error messages). 

I am glad that this issue is marked complete. Any of the high priority issues resolved is great news!

Hopefully the manifest issue can be marked resolved soon too.
